manifests/init.pp

```diff
@@ -1,29 +1,31 @@
-# Manage SELinux state
+# Manage active SELinux state and state after a reboot
 #
 # @param ensure
 #   The state that SELinux should be in.
-#   Valid values are: true, false, 'enforcing', 'permissive', 'disabled'.
 #   Since you are calling this class, we assume that you want to enforce.
 #
 # @param mode
 #   The SELinux type you want to enforce.
-#   Valid values are: 'targeted', 'mls'
 #   Note, it is quite possible that 'mls' will render your system inoperable.
 #
 # @param manage_utils_package
 #   If true, ensure policycoreutils-python is installed. This is a supplemental
 #   package that is required by semanage.
 #
-# Additional functionality for SELinux support.
-#
 class selinux (
-  Variant[Boolean,Enum['enforcing','permissive','disabled']] $ensure = 'enforcing',
+  Selinux::State         $ensure               = simplib::lookup('simp_options::selinux', { 'default_value' => true }),
```

Comment:
This should also handle `enforcing`, `permissive`, and `disabled` so that it matches user expectations.